src/wp-admin/includes/class-theme-upgrader.php

```diff
@@ -566,15 +566,27 @@ public function check_package( $source ) {
 			);
 		}
 
-		// If it's not a child theme, it must have at least an index.php to be legit.
-		if ( empty( $info['Template'] ) && ! file_exists( $working_directory . 'index.php' ) ) {
+		/*
+		 * Parent themes must contain an index file:
+		 * - classic themes require /index.php
+		 * - block themes require /templates/index.html
+		 */
+		if (
+			empty( $info['Template'] ) &&
+			! file_exists( $working_directory . 'index.php' ) &&
+			! file_exists( $working_directory . 'templates/index.html' )
+		) {
```

Comment:
This has changed the behavior for classic themes. Before this PR, a classic theme without a `index.php` file could not be uploaded. After this PR, it can be if it has a `templates/index.html` file. Is this problematic?

Reply:
I'm not sure, I based it on the code for activating plugins in [r52940](https://core.trac.wordpress.org/changeset/52940). 

My reading of the [docs for block themes](https://developer.wordpress.org/themes/block-themes/block-theme-setup/) is that `template/index.html` is what defines a block theme. Is there something else the code could be checking for?

Reply:
Yes it is for block themes. But non-block themes could also have a `template` directory and an `index.html` file within it.

My concern (though not tested or validated) is: does this allow a non-block theme to be installed but then not work elsewhere in the admin area or frontend? Hmm, need to test this to find out.

Reply:
A classic theme cannot have a `templates/index.html` since `WP_Theme::is_block_theme()` was introduced in 5.9. As soon as it has that file, it's considered a block theme.